misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java

```diff
@@ -81,14 +83,24 @@ public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> an
         if (!enabled) {
             return;
         }
-        String packageName = getPackageName((BLangPackage) ((BLangFunction) functionNode).parent);
+        BLangPackage bLangPackage = (BLangPackage) ((BLangFunction) functionNode).parent;
+        String packageName = getPackageName(bLangPackage);
         suite = registry.getTestSuites().get(packageName);
         // Check if the registry contains a test suite for the package
         if (suite == null) {
-            // Add a test suite to the registry if it does not contain one pertaining to the package name
-            registry.getTestSuites().computeIfAbsent(packageName, func -> new TestSuite(packageName));
-            // Get the test suite related to the package from registry
-            suite = registry.getTestSuites().get(packageName);
+            //Set testable flag for single bal file execution
+            if ((Names.DOT.getValue()).equals(packageName)) {
```

Comment:
Are we checking Testable flag any other place? if not we can combine both if conditions

Reply:
No we don't check that flag anywhere else. I'll combine the conditions.